nannyml/performance_estimation/confidence_based/_cbpe_binary_classification.py

```diff
@@ -104,6 +104,8 @@ def __init__(
         )
 
         self.model_metadata = model_metadata  # seems to be required for typing to kick in
+        self.metric_upper_bound = 1
```

Comment:
Was doubting naming this confidence upper bound (since it is the combination of metric+confidence that should be below 1). That way it is also clear what this variable is being used for.

Don't feel very strongly about it.

Reply:
Also I don't mind having these. I always prefer having constants defined up front with a clear name then just putting a magical `1` or `0` value somewhere.